scapy/supersocket.py

```diff
@@ -75,12 +75,12 @@ def close(self):
         if self.closed:
             return
         self.closed = True
-        if hasattr(self, "outs"):
-            if not hasattr(self, "ins") or self.ins != self.outs:
-                if self.outs and (WINDOWS or self.outs.fileno() != -1):
+        if getattr(self, "outs", None):
+            if getattr(self, "ins", self.outs) != self.outs:
```

Comment:
If there is no `ins` argument, this will be false, and `self.outs` won't be closed. Or did I miss something?
```suggestion
            if getattr(self, "ins", None) != self.outs:
```